runtime/caml/mlvalues.h

```diff
@@ -91,6 +91,9 @@ typedef opcode_t * code_t;
 #define Is_exception_result(v) (((v) & 3) == 2)
 #define Extract_exception(v) ((v) & ~3)
 
+/* or_null */
+#define Val_null ((value) 0)
```

Comment:
As far as I can tell, every bit of code in the runtime that handles an arbitrary OCaml value and checks `Is_block(v)` now needs to check `v != Val_nul && Is_block(v)`. Should we just integrate these into the `Is_block / Is_long` macros? `Val_null` is more like an immediate than a pointer, and making `Is_block(Val_null)` false breaks fewer invariants. (For instance, much code assumes that `Tag_val(v)` is defined if `Is_block(v)` holds).

Reply:
I'm unsure about this, let's discuss next week.

Reply:
I'm also somewhat worried about the performance impact. Perhaps we can skip the null pointer checks in some places.

Reply:
I think that having a gap between `Is_long` and `Is_block` is more error-prone, and I'd prefer that `Is_long(Val_null)` was true.

You should make `Is_block` a static inline function rather than a macro to avoid the issue you point out.

Reply:
Done.